sale_stock_medical_prescription/tests/test_sale_order_line.py

```diff
@@ -0,0 +1,129 @@
+        self.order_line_13.product_uom = dozen
+        self.assertEquals(
+            self.order_line_13.dispense_qty,
+            self.order_line_13.product_uom_qty * 12,
+        )
+
+    def test_check_product(self):
+        """ Test changing product to incorrect one raise ValidationError """
+        product = self.env['product.product'].search([
+            ('id', '!=', self.order_line_12.product_id.id)
+        ], limit=1)
+        with self.assertRaises(ValidationError):
+            self.order_line_12.product_id = product.id
+
+    def test_check_can_dispense_false(self):
+        """ Test raise ValidationError if try to dispense but cannot """
+        with self.assertRaises(ValidationError):
+            self.order_line_12._check_can_dispense()
+
+    def test_check_can_dispense_conditions_all_true(self):
+        """ Test no Validation Errors if NOT all conditions True """
+        self.order_line_12.product_id.is_medicament = False
+        try:
+            self.order_line_12._check_can_dispense()
+            self.assertTrue(True)
+        except ValidationError:
+            self.fail()
+
+    def test_check_can_dispense_qty_greater_than_rx_line_can_dispense(self):
+        """ Test raise ValidationError if try dispense but not enough stock """
+        self.order_line_13.product_uom_qty = 1000
+        with self.assertRaises(ValidationError):
+            self.order_line_13._check_can_dispense()
+
+    def test_prepare_order_line_procurement_otc(self):
+        """ It should not throw an error and should assign OTC route """
+        self.amoxicillin.is_prescription = False
+        res = self.order_line_12._prepare_order_line_procurement()
+        expect = self.env.ref(
+            'sale_stock_medical_prescription.route_warehouse0_otc'
+        )
+        res = res['route_ids'][0][2]
+        self.assertEquals(
+            [expect.id], res,
+            'Did not correctly assign route. Expect %s, Got %s' % (
+                [expect.id], res
+            )
+        )
+
+    def test_prepare_order_line_procurement_prescription(self):
+        """ It should not throw an error and should assign RX route """
+        res = self.order_line_12._prepare_order_line_procurement()
+        expect = self.env.ref(
+            'sale_stock_medical_prescription.route_warehouse0_prescription'
+        )
+        res = res['route_ids'][0][2]
+        self.assertEquals(
+            [expect.id], res,
+            'Did not correctly assign route. Expect %s, Got %s' % (
+                [expect.id], res
+            )
+        )
+
+    def test_check_can_dispense_no_order_lines(self):
+        """ Test Validations skipped if no prescription lines present """
+        try:
+            self.order_line_12.prescription_order_line_id = None
+            self.assertTrue(True)
+        except ValidationError:
+            self.fail(
+                'Should skip validations if no rx_line.'
+            )
+
+    def test_check_patient(self):
```

Comment:
@lasley I re-investigated this and found the following when running test_check_patient here (with test_check_patient in rx_line commented out.)
* Steps taken for _check_patient
     * Test _check_patient in sale order line, set its patient to patient_2 (demo from medical module)
        * If I have _check_patient in both sale_order_line and rx_line
            * _check_patient in rx_line model only runs, and it doesn't raise a validation error because even though I changed the patient only on the sale_order_line, both the sale_order_line's and rx_lines's patients are set to the same, new patient automatically. And thus the validation error is not triggered.
        * If I comment out the rx_line's _check_patient
            * When the _check_patient runs in the sale_order_line model, it exits because there is no medical_prescription_order_line attached to it. I checked the demo data, and I explicitly set an rx_line on it and it is also present in the UI.